scanner_test.go

```diff
@@ -91,9 +91,11 @@ func TestScanner_Scan(t *testing.T) {
 		AssertScan(t, `123.E45`, sql.FLOAT, `123.E45`)
 		AssertScan(t, `123E+4`, sql.FLOAT, `123E+4`)
 		AssertScan(t, `123E-4`, sql.FLOAT, `123E-4`)
+		AssertScan(t, `.0E-2`, sql.FLOAT, `.0E-2`)
 		AssertScan(t, `123E`, sql.ILLEGAL, `123E`)
 		AssertScan(t, `123E+`, sql.ILLEGAL, `123E+`)
 		AssertScan(t, `123E-`, sql.ILLEGAL, `123E-`)
+		AssertScan(t, `.E2`, sql.DOT, `.`)
```

Comment:
This does seem like an important test, but it's the wrong place for it, right? This is a test for testing the scan of floats, but this is not a float. Is `.E2` meant to be scanned as a type of token? If so, what type?

Reply:
I moved it into the `DOT` subtest